main.go

```diff
@@ -129,44 +257,75 @@ func (c *CASProxy) ValidateTicket(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	username := string(fields[1])
+
 	// Store a session, hopefully to short-circuit the CAS redirect dance in later
 	// requests. The max age of the cookie should be less than the lifetime of
 	// the CAS ticket, which is around 10+ hours. This means that we'll be hitting
 	// the CAS server fairly often. Adjust the max age to rate limit requests to
 	// CAS.
-	err = session.PutInt(r, sessionKey, 1)
+	var s *sessions.Session
+	s, err = c.sessionStore.Get(r, sessionName)
 	if err != nil {
-		err = errors.Wrap(err, "error setting setting value")
+		err = errors.Wrap(err, "error getting session")
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	s.Values[sessionKey] = username
+	s.Save(r, w)
 
 	http.Redirect(w, r, svcURL.String(), http.StatusFound)
 }
 
+// ResetSessionExpiration should reset the session expiration time.
+func (c *CASProxy) ResetSessionExpiration(w http.ResponseWriter, r *http.Request) error {
+	session, err := c.sessionStore.Get(r, sessionName)
+	if err != nil {
+		return err
+	}
+
+	msg, ok := session.Values[sessionKey]
+	if !ok {
+		return errors.New("session value not found")
+	}
+
+	session.Values[sessionKey] = msg.(string)
+	session.Save(r, w)
+	return nil
+}
+
 // Session implements the mux.Matcher interface so that requests can be routed
 // based on cookie existence.
 func (c *CASProxy) Session(r *http.Request, m *mux.RouteMatch) bool {
-	msg, err := session.GetInt(r, sessionKey)
+	session, err := c.sessionStore.Get(r, sessionName)
 	if err != nil {
 		return true
 	}
 
-	if msg != 1 {
-		log.Infof("session value was %d instead of 1", msg)
+	msgraw, ok := session.Values[sessionKey]
```

Comment:
I fell like maybe this should be `msgRaw` or we'll be eternally confused, if momentarily